Tests/MongoSwiftTests/BSONValueTests.swift

```diff
@@ -27,4 +28,89 @@ final class BSONValueTests: XCTestCase {
         expect(try Binary(data: sixteenBytes, subtype: .uuidDeprecated)).toNot(throwError())
         expect(try Binary(data: sixteenBytes, subtype: .uuid)).toNot(throwError())
     }
+
+    func bsonEqual(expectedValue: BSONValue?) -> Predicate<BSONValue> {
+        return Predicate { (actualExpression: Expression<BSONValue>) throws -> PredicateResult in
+            let msg = ExpectationMessage.expectedActualValueTo("equal <\(String(describing: expectedValue)))>")
+            if let actualValue = try actualExpression.evaluate() {
+                return PredicateResult(
+                    bool: try bsonEquals(lhs: actualValue, rhs: expectedValue!),
+                    message: msg
+                )
+            } else {
+                return PredicateResult(
+                    status: .fail,
+                    message: msg.appendedBeNilHint()
+                )
+            }
+        }
+    }
+
+    func checkTrueAndFalse(val: BSONValue, alternate: BSONValue) {
+        expect(val).to(bsonEqual(expectedValue: val))
+        expect(val).toNot(bsonEqual(expectedValue: alternate))
+    }
+
+    func testBSONEquals() throws {
+        // Int
+        checkTrueAndFalse(val: 1, alternate: 2)
+        // Int32
+        checkTrueAndFalse(val: Int32(32), alternate: Int32(33))
+        // Int64
+        checkTrueAndFalse(val: Int64(64), alternate: Int64(65))
+        // Double
+        checkTrueAndFalse(val: 1.618, alternate: 2.718)
+        // Decimal128
+        checkTrueAndFalse(val: Decimal128("1.618"), alternate: Decimal128("2.718"))
+        // Bool
+        checkTrueAndFalse(val: true, alternate: false)
+        // String
+        checkTrueAndFalse(val: "some", alternate: "not some")
+        // RegularExpression
+        checkTrueAndFalse(
+            val: RegularExpression(pattern: ".*", options: ""),
+            alternate: RegularExpression(pattern: ".+", options: "")
+        )
+        // Timestamp
+        checkTrueAndFalse(val: Timestamp(timestamp: 1, inc: 2), alternate: Timestamp(timestamp: 5, inc: 10))
+        // Date
+        checkTrueAndFalse(
+            val: Date(timeIntervalSinceReferenceDate: 5000),
+            alternate: Date(timeIntervalSinceReferenceDate: 5001)
+        )
+        // MinKey & MaxKey
+        expect(MinKey()).to(bsonEqual(expectedValue: MinKey()))
+        expect(MaxKey()).to(bsonEqual(expectedValue: MaxKey()))
```

Comment:
Any reason not to use `checkTrueAndFalse()` here? I think we're skipping the `false` case for both types as-is.

Reply:
`MinKey` and `MaxKey` are both... interesting types. The `MinKey` type only ever has a single value, so the equality operation on it is essentially vacuously true. This of course also applies to `MaxKey`.

So, testing something like `MaxKey =?= MinKey` or vice-versa, is really just testing the differing types case, which is tested in `BSONValueTests.swift` in the last few lines:
```
// Different types
 expect(4).toNot(bsonEqual(expectedValue: "swift"))
```